src/modules/admin/components/Tokens/TokensMoveDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback } from 'react';
+import { FormikProps } from 'formik';
+import * as yup from 'yup';
+import moveDecimal from 'move-decimal-point';
+import BigNumber from 'bn.js';
+
+import { ZERO_ADDRESS } from '~utils/web3/constants';
+import { pipe, mapPayload, withKey } from '~utils/actions';
+import { Address } from '~types/index';
+import { ActionTypes } from '~redux/index';
+import Dialog from '~core/Dialog';
+import { ActionForm } from '~core/Fields';
+
+import { useColonyTokens } from '../../../dashboard/hooks/useColonyTokens';
+import DialogForm from './TokensMoveDialogForm';
+
+export interface FormValues {
+  fromDomain?: number;
+  toDomain?: number;
+  amount: string;
+  tokenAddress?: Address;
+}
+
+interface Props {
+  cancel: () => void;
+  close: (params: object) => void;
+  colonyAddress: Address;
+  toDomain?: number;
+}
+
+const TokensMoveDialog = ({
+  colonyAddress,
+  toDomain,
+  cancel,
+  close,
+}: Props) => {
+  const validationSchema = yup.object().shape({
+    fromDomain: yup.string().required(),
+    toDomain: yup.string().required(),
+    amount: yup.number().required(),
+    tokenAddress: yup.string().required(),
+  });
+
+  // Fetch colony tokens here since we need them for form submission
+  const [colonyTokenRefs, colonyTokens] = useColonyTokens(colonyAddress);
+
+  const transform = useCallback(
+    pipe(
+      mapPayload(payload => {
+        // Find the selected token's decimals
+        const { decimals: tokenDecimals } = (colonyTokens &&
+          colonyTokens.find(
+            token => token.address === payload.tokenAddress,
+          )) || {
+          decimals: undefined,
+        };
+
+        // Convert amount string with decimals to BigInt (eth to wei)
+        const amount = new BigNumber(
+          moveDecimal(payload.amount, tokenDecimals || 18),
+        );
+
+        return { ...payload, colonyAddress, amount };
+      }),
+      withKey(colonyAddress),
+    ),
+    [colonyAddress, colonyTokens],
+  );
+
+  return (
+    <ActionForm
+      initialValues={{
+        fromDomain: undefined,
+        toDomain,
+        amount: '',
+        tokenAddress: ZERO_ADDRESS,
+      }}
+      validationSchema={validationSchema}
+      submit={ActionTypes.MOVE_FUNDS_BETWEEN_POTS}
+      error={ActionTypes.MOVE_FUNDS_BETWEEN_POTS_ERROR}
+      // Close dialog immediately to give way for GasStation
+      success={ActionTypes.MOVE_FUNDS_BETWEEN_POTS}
```

Comment:
We display no pending state whatsoever, right ? Like, it wasn't spec'd by product ?

Ie: The move funds dialog closes, the gas station opens (be we close it). Is there any way to tell there's a pending action ?